build/lib/src/generate/run_builder.dart

```diff
@@ -62,3 +93,26 @@ Future<void> runBuilder(Builder builder, Iterable<AssetId> inputs,
     await resources.beforeExit();
   }
 }
+
+extension on Package {
+  Package transformToAssetUris() {
+    return Package(
+      name,
+      Uri(scheme: 'asset', pathSegments: [name, '']),
+      packageUriRoot: Uri(scheme: 'asset', pathSegments: [name, 'lib', '']),
+      relativeRoot: false,
```

Comment:
This could be a bit more general?

```suggestion
      packageUriRoot: packageUriRoot,
      relativeRoot: relativeRoot,
```